src/NUnitFramework/framework/Constraints/AttributeConstraint.cs

```diff
@@ -57,6 +58,10 @@ public AttributeConstraint(Type type, IConstraint baseConstraint)
         /// expected attribute and if its value matches the
         /// additional constraint specified.
         /// </summary>
+        /// <typeparam name="TActual"></typeparam>
+        /// <param name="actual">The value to be tested</param>
+        /// <exception cref="ArgumentException">Attribute was not found</exception>
+        /// <returns></returns>
         public override ConstraintResult ApplyTo<TActual>(TActual actual)
         {
             Guard.ArgumentNotNull(actual, "actual");
```

Comment:
`Guard.ArgumentNotNull` may throw an `ArgumentNullException`, so you might want to add that too. Probably plenty of other cases if you are feeling industrious :smile: